traitlets/traitlets.py

```diff
@@ -2477,8 +2477,9 @@ def instance_init(self, obj):
 class Dict(Instance):
     """An instance of a Python dict."""
     _trait = None
+    _key_trait = None
 
-    def __init__(self, trait=None, traits=None, default_value=Undefined,
+    def __init__(self, trait=None, key_trait=None, traits=None, default_value=Undefined,
```

Comment:
To be honest, I think this schema feature accessible via the `traits` argument doesn't belong into `Dict` at all. I'd rather not have it block access to `key_trait`. What about a compromise: Keep `key_trait` before `traits` and when an instance of `dict` (the python class, not the trait) is encountered in `key_trait`, raise a warning and map it to `traits`?